src/mscorlib/src/System/String.cs

```diff
@@ -159,27 +159,26 @@ public sealed class String : IComparable, ICloneable, IConvertible, IEnumerable
             Contract.Ensures(Contract.Result<String>() != null);
             Contract.EndContractBlock();
 
-            if (separator == null)
-                separator = String.Empty;
-
-
             using(IEnumerator<String> en = values.GetEnumerator()) {
                 if (!en.MoveNext())
                     return String.Empty;
 
-                StringBuilder result = StringBuilderCache.Acquire();
-                if (en.Current != null) {
-                    result.Append(en.Current);
-                }
+                String value = en.Current;
 
-                while (en.MoveNext()) {
-                    result.Append(separator);
-                    if (en.Current != null) {
+                if (en.MoveNext()) { 
+                    StringBuilder result = StringBuilderCache.Acquire();
+                    result.Append(value);
+
+                    do {
+                        result.Append(separator);
                         result.Append(en.Current);
-                    }
-                }            
-                return StringBuilderCache.GetStringAndRelease(result);
-            }           
+                    } while (en.MoveNext());
+                    return StringBuilderCache.GetStringAndRelease(result);
+                }
+
+                // Only one value available
+                return value ?? String.Empty;
```

Comment:
We did not use '??' in this file before. Can you use the ?: for consistency's sake?